docs/excel/show-hide-add-in.md

```diff
@@ -0,0 +1,77 @@
+
+```javascript
+function onCurrentQuarter() {
+    Office.addin.showAsTaskpane()
+    .then(function() {
+        // Code that enables task pane UI elements for
+        // working with the current quarter.
+    });
+}
+```
+
+To hide the task pane, your code calls `Office.addin.hide()`. The following example is a handler that is registered for the [Office.Worksheet.onDeactivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview#ondeactivated) event.
+
+```javascript
+function onCurrentQuarterDeactivated() {
+    Office.addin.hide();
+}
+```
+
+### Preservation of state and event listeners
+
+The `hide()` and `showAsTaskpane()` methods only change the *visibility* of the task pane. They do not unload or reload it (or reinitialize its state).
+
+Consider the following scenario: A task pane is designed with tabs. The **Home** tab is open when the add-in is first launched. Suppose a user opens the **Settings** tab and, later, code in the task pane calls `hide()` in response to some event. Still later code calls `showAsTaskpane()` in response to another event. The task pane will reappear, and the **Settings** tab is still elected.
+
+![A screenshot of task pane that has four tabs labelled Home, Settings, Favorites, and Accounts.](../images/TaskpaneWithTabs.png)
+
+In addition, any event listeners that are registered in the task pane continue to run even when the task pane is hidden.
+
+Consider the following scenario: The task pane has a registered handler for the Excel `Worksheet.onActivated` and `Excel.Worksheet.onDeactivated` events for a sheet named **Sheet1**. The activated handler causes a green dot to appear in the task pane. The deactivated handler turns the dot red (which is its default state). Suppose then that code calls `hide()` when **Sheet1** is not activated and the dot is red. While the task pane is hidden, **Sheet1** is activated. Later code calls `showAsTaskpane()` in response to some event. When the task pane opens, the dot is green because the event listeners and handlers ran even though the task pane was hidden.
+
+### Handle visibility changed event
+
+Changing the visibility of the task pane is an event that your code can handle. To register a handler for the event, you do not use an "add handler" method as you would in most Office JavaScript contexts. Instead, there is a special function to which you pass your handler: `Office.addin.onVisibilityModeChanged`. The function returns another function that deregisters the handler. The following is an example. Note that the `visibilityMode` property of the `args` object that Office passes to the handler can have the values "Hidden" or "Taskpane".
+
+```javascript
+var removeVisibilityModeHandler =
+    await Office.addin.onVisibilityModeChanged(function(args) {
+        console.log('Visibility is changed to ' + args.visibilityMode);
+});
+```
+
+To deregister the handler:
+
+```javascript
+await removeVisibilityModeHandler();
```

Comment:
Are we officially okay with having `await` in the JS examples? If so, maybe we should consider cleaning up some older `then()` examples.

Reply:
rewrote whole section